code/modules/mining/boulder_processing/brm.dm

```diff
@@ -0,0 +1,143 @@
+
+/obj/machinery/bouldertech/brm/process()
+	if(SSore_generation.available_boulders.len < 1)
+		say("No boulders to collect. Entering idle mode.")
+		toggled_on = FALSE
+		STOP_PROCESSING(SSmachines, src)
+		update_appearance(UPDATE_ICON_STATE)
+		return
+	for(var/i in 1 to boulders_processing_max)
+		if(!pre_collect_boulder())
+			toggled_on = FALSE
+			STOP_PROCESSING(SSmachines, src)
+			update_appearance(UPDATE_ICON_STATE)
+	for(var/ground_rocks in loc.contents)
+		if(istype(ground_rocks, /obj/item/boulder))
+			boulders_contained += ground_rocks
+			if(boulders_contained.len > boulders_held_max)
+				toggled_on = FALSE
+				STOP_PROCESSING(SSmachines, src)
+				boulders_contained = list()
+				update_appearance(UPDATE_ICON_STATE)
+				return
+
+/obj/machinery/bouldertech/brm/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	context[SCREENTIP_CONTEXT_LMB] = "Teleport single boulder"
+	context[SCREENTIP_CONTEXT_RMB] = "Toggle automatic boulder retrieval"
+	return CONTEXTUAL_SCREENTIP_SET
+
+/obj/machinery/bouldertech/brm/examine(mob/user)
+	. = ..()
+	. += span_notice("The small screen reads there are [span_boldnotice("[SSore_generation.available_boulders.len] boulders")] available to teleport.")
+
+/obj/machinery/bouldertech/brm/RefreshParts()
+	. = ..()
+	var/scanner_stack = 0
+	var/laser_stack = 0
+	for(var/datum/stock_part/scanning_module/scanner in component_parts)
+		scanner_stack += (scanner.tier)
+	boulders_processing_max = scanner_stack
+	for(var/datum/stock_part/micro_laser/laser in component_parts)
+		laser_stack += (laser.tier)
+	boulders_held_max = laser_stack + 1
+
+/obj/machinery/bouldertech/brm/update_icon_state()
+	if(toggled_on && !panel_open)
+		icon_state = "[initial(icon_state)]-toggled"
+		return
+	return ..()
+
+/obj/machinery/bouldertech/brm/proc/pre_collect_boulder()
+	if(!SSore_generation.available_boulders.len)
+		playsound(loc, 'sound/machines/synth_no.ogg', 30 , TRUE)
+		balloon_alert_to_viewers("no boulders to collect!")
+		return FALSE //Nothing to collect
+	var/obj/item/boulder/random_boulder = pick(SSore_generation.available_boulders)
+	if(random_boulder.processed_by)
+		return FALSE
+	if(!random_boulder)
+		return FALSE
+	random_boulder.processed_by = src
+	random_boulder.Shake(duration = 1.5 SECONDS)
+	SSore_generation.available_boulders -= random_boulder
+	addtimer(CALLBACK(src, PROC_REF(collect_boulder), random_boulder), 1.5 SECONDS)
+	return TRUE
+
+/obj/machinery/bouldertech/brm/proc/collect_boulder(obj/item/boulder/random_boulder)
+	flick("brm-flash", src)
+	if(QDELETED(random_boulder))
+		playsound(loc, 'sound/machines/synth_no.ogg', 30 , TRUE)
+		balloon_alert_to_viewers("target lost!")
+		return FALSE
+	playsound(src, AUTO_TELEPORT_SOUND, 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	random_boulder.forceMove(drop_location())
+	balloon_alert_to_viewers("boulder appears!")
+	random_boulder.visible_message(span_warning("[random_boulder] suddenly appears!"))
+	use_power(100)
```

Comment:
`100` looks like a radom value. we usually use `active_power_usage` var. If you want it to be 100 then do `active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * 0.1` and then use that